app/src/main/java/org/mozilla/focus/urlinput/UrlInputPresenter.java

```diff
@@ -167,5 +156,34 @@ protected void onPostExecute(List<CharSequence> strings) {
 
             return suggests;
         }
+
+        private static String readLines(URLConnection connection) throws IOException {
+            InputStream inputStream;
+            try {
+                inputStream = connection.getInputStream();
+            } catch (IndexOutOfBoundsException ignored) {
+                // IndexOutOfBoundsException sometimes is thrown by the okHttp library
+                // bundled within the android framework, we can only catch the exception here,
+                // or use the latest okHttp3.
+                return "";
+            }
+
+            BufferedReader bufferedReader = null;
+            StringBuilder total = new StringBuilder();
+            try {
+                InputStreamReader reader = new InputStreamReader(new BufferedInputStream(inputStream),
+                        "utf-8");
+                bufferedReader = new BufferedReader(reader);
+
+                String line;
+                while ((line = bufferedReader.readLine()) != null) {
+                    total.append(line).append('\n');
+                }
+            } finally {
+                IOUtils.closeQuietly(bufferedReader);
```

Comment:
Consider rework this part we Autoclosabe: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html